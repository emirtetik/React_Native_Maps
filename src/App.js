import React,{useRef,useState} from 'react'
import { View,Text,TextInput,FlatList,StyleSheet } from 'react-native'
import Config from 'react-native-config';
import MapView from 'react-native-maps';
import useFetch from './hooks/useFetch';
import Loading from './components/Loading/Loading';
import UserMarker from './components/marker/UserMarker/UserMarker';
import InfoCard from './components/Modal/infoCard/InfoCard';

API_URL="https://random-data-api.com/api/v2/users?size=30"


const App = ( ) => {
  const mapRef = useRef()
  const {data,loading,error} = useFetch(API_URL);
  const [user,setUser] = useState();
  const [infovisible,setınfoVisible] = useState(false);


   
  const renderUserMarkers = () => {
   return data.map(({id,first_name,last_name,username, avatar, address:{coordinates}}) => {
      return(
        <UserMarker
        key={id}
        coordinates={
          {
           latitude:coordinates.lat,
           longitude:coordinates.lng,
          }
        }
        userImage={avatar}
        onSelect={() => handleMarkerSelect(coordinates,{first_name,last_name,username})}
        
        />
      )
     })
  }

  function handleMarkerSelect(coor,selectedUser) {
    setUser(selectedUser)
    handleModalVisible();
    mapRef.current.animateToRegion({
        latitude: coor.lat,
        longitude: coor.lng,
        latitudeDelta: 5,
        longitudeDelta: 5,
    })
  }
 
  function handleModalVisible() {
     setınfoVisible(!infovisible)
  }
     return(

        <View style={styles.container}>

        <MapView ref={mapRef}  style={styles.map}>
       {data && renderUserMarkers()}
        </MapView>

        {loading && <Loading />}

   { user &&  (  <InfoCard visible={infovisible} onClose={handleModalVisible}  user={user}/>
   )
 }

        </View>
     )
}

const styles = StyleSheet.create({
  container:{
     flex: 1,
  },
  map:{
    flex:1,
  },
 header:{
  fontSize: 24,
  margin:10
 }
 
})
export default App



















//  const renderElements = ( {item}) => <Text>{item}</Text>;
  
//  const [text, setText] = React.useState("")

//  const [addList, setAddList] = React.useState([])
 
//  function addTolist() {
//    if(!text){
//     return;
//    }
//   setAddList([...addList, text]);
  
//  }


      {/* <Text style={styles.header}>   Testingasdasfd  </Text> */}
{/* 
      <FlatList 
      data={[addList]} 
      renderItem={renderElements} 
      testID='list' 
      keyExtractor={(_, index) => index.toString()}/>

      <TextInput
      placeholder='Add...'  
      style={styles.input} 
      onChangeText={setText} 
      testID='Change-text'/>

       <Button 
       title="Add" 
       onClick={addTolist} /> */}


        // input:{
  //   backgroundColor:"#bdbdbdbd",
  //   padding:10,
  //   margin:10,
  //   borderWidth:1,
  //   borderColor:"gray",
  //   borderRadius:8,
  // }