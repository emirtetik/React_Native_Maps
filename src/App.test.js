import {render, fireEvent} from '@testing-library/react-native';
import React from 'react';
import App from './App';

test('should match with snapshot', () => {
  const comp = render(<App />);
  expect(comp).toMatchSnapshot()
});

test('should add item to list', () => {
  const comp = render(<App />);

  const input = comp.getByTestId('Change-text');
  fireEvent.changeText(input, 'test element');

  const submitButton = comp.getByTestId('touchable-button');
  fireEvent.press(submitButton);

  const list = comp.getByTestId('list').props;
  expect(list.data.length).toBe(1);
});


test('should add item for multiple times', () => { 

    const comp = render(<App />);

    const input = comp.getByTestId("Change-text")
  fireEvent.changeText(input, 'test element');

   const submitButton = comp.getByTestId("touchable-button");
   fireEvent.press(submitButton)
   fireEvent.press(submitButton)
   fireEvent.press(submitButton)

   const list = comp.getByTestId('list').props;
  expect(list.data.length).toBe(3);
 })


 
// test('should not add item if input are empty ', () => { 

//     const comp = render(<App />);


//    const submitButton = comp.getByTestId("touchable-button");
//    fireEvent.press(submitButton)

//    const list = comp.getByTestId('list').props;
//   expect(list.data.length).toBe(0);
//  })