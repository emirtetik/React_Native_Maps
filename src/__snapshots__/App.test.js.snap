// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match with snapshot 1`] = `
<View
  style={
    {
      "flex": 1,
    }
  }
>
  <Text
    style={
      {
        "color": "black",
        "fontSize": 30,
      }
    }
  >
       Testing  
  </Text>
  <RCTScrollView
    data={
      [
        [],
      ]
    }
    getItem={[Function]}
    getItemCount={[Function]}
    keyExtractor={[Function]}
    onContentSizeChange={[Function]}
    onLayout={[Function]}
    onMomentumScrollBegin={[Function]}
    onMomentumScrollEnd={[Function]}
    onScroll={[Function]}
    onScrollBeginDrag={[Function]}
    onScrollEndDrag={[Function]}
    removeClippedSubviews={false}
    renderItem={[Function]}
    scrollEventThrottle={50}
    stickyHeaderIndices={[]}
    testID="list"
    viewabilityConfigCallbackPairs={[]}
  >
    <View>
      <View
        onFocusCapture={[Function]}
        onLayout={[Function]}
        style={null}
      >
        <Text />
      </View>
    </View>
  </RCTScrollView>
  <TextInput
    onChangeText={[Function]}
    placeholder="Add..."
    style={
      {
        "backgroundColor": "#bdbdbdbd",
        "borderColor": "gray",
        "borderRadius": 8,
        "borderWidth": 1,
        "margin": 10,
        "padding": 10,
      }
    }
    testID="Change-text"
  />
  <View
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": undefined,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "backgroundColor": "blue",
        "borderColor": "balck",
        "borderRadius": 8,
        "borderWidth": 1,
        "margin": 10,
        "opacity": 1,
        "padding": 10,
      }
    }
    testID="touchable-button"
  >
    <Text
      style={
        {
          "color": "white",
        }
      }
    >
      Add
    </Text>
  </View>
</View>
`;
